pandas/tests/frame/test_constructors.py

```diff
@@ -2396,6 +2396,15 @@ def check_views():
             # assert b[0] == 0
             assert df.iloc[0, 2] == 0
 
+    def test_consistency_of_string_columns_with_none(self):
+        # df created from list, None is casted to str
+        df = DataFrame(["1", "2", None], columns=["a"], dtype="str")
+        type(df.loc[2].values[0])
```

Comment:
so this is not actually testing anything, instead you need to actually construct the expected frame and use assert_frame_equal. more to the point, I am pretty sure we have tests very similar to this. happy to take if we don't, but pls have a look (would take this if its slightly different than what we have, so pls give a look). we have a ton of tests so have to look around a bit.

Reply:
Hey @jreback, by accident I didn't push the next commit with some actual asserts. 
However, if there is already a similar test that resolves #32218 we can reference it and close this PR.

Reply:
yeah if you'd have a look around. I recall a similar issue to this but nothing referenced, pls have a look see at the existing tests (I know there are a lot)

Reply:
It was fixed in 1.2.0 (but broken in 1.1.5), but I don't directly find anything in the 1.2.0 notes about this (and also didn't directly find an issue about it). So I think it is fine to add the test to be safe.